cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ScheduledRunRecordCorrectorService.java

```diff
@@ -85,13 +107,23 @@ protected void shutDown() throws Exception {
 
     @Override
     public void run() {
+      if (done) {
```

Comment:
Why do we need this check if the `scheduledExecutorService` is shut down when we set it?
It is in the case that the `scheduledExecutorService` takes longer than 3 seconds to shut down (3, being the interval between runs)?

Reply:
yes that was the idea. In case the ```scheduledExecutorService``` takes longer to shut down - because it gets shut down from the scheduled task itself, I did not want to ```awaitTermination()```. so there is a slight chance that it does not shutdown.